src/server/game/Entities/Player/Player.cpp

```diff
@@ -12648,18 +12652,23 @@ void Player::StoreLootItem(uint8 lootSlot, Loot* loot)
 
 uint32 Player::CalculateTalentsPoints() const
 {
+    uint32 talentPointsForLevel = 0;
     uint32 base_talent = getLevel() < 10 ? 0 : getLevel() - 9;
 
     if (getClass() != CLASS_DEATH_KNIGHT || GetMapId() != 609)
-        return uint32(base_talent * sWorld->getRate(RATE_TALENT));
-
-    uint32 talentPointsForLevel = getLevel() < 56 ? 0 : getLevel() - 55;
-    talentPointsForLevel += m_questRewardTalentCount;
-
-    if (talentPointsForLevel > base_talent)
+    {
         talentPointsForLevel = base_talent;
+    }
+    else
+    {
+        talentPointsForLevel = getLevel() < 56 ? 0 : getLevel() - 55;
```

Comment:
I suggest to assign getLevel() to a variable and do not call it twice.
```suggestion
        auto playerLevel = getLevel();
        talentPointsForLevel = playerLevel < 56 ? 0 : playerLevel - 55;
```

Reply:
looking better at the code, you can create `auto playerLevel = getLevel();` at the begin of this function and avoid calling getLevel() multiple times.